google/cloud/spanner_dbapi/cursor.py

```diff
@@ -258,9 +259,21 @@ def executemany(self, operation, seq_of_params):
 
         many_result_set = StreamedManyResultSets()
 
-        for params in seq_of_params:
-            self.execute(operation, params)
-            many_result_set.add_iter(self._itr)
+        if classification == parse_utils.STMT_INSERT:
```

Comment:
If I understand this correctly, this method will now use mutations for all insert statements, and DML for all other update statements (`UPDATE`, `DELETE`). I think we should try to make this in a slightly different way, so users can have some more control over it, and it can be used for update and delete as well. I would suggest:
1. Use mutations by default when the connection is in auto-commit mode. Reason: Mutations do not support read-your-writes, but that is not possible in auto-commit mode anyways.
2. Use DML by default when the connection is in transactional mode (for the same reason as above).
3. Let the user configure whether a specific transaction should use DML or mutations. This could be through a custom method or a custom SQL statement. As far as I know, dbapi does not have a standard method for setting the isolation level for transactions. That is the method that we [used for ActiveRecord](https://github.com/googleapis/ruby-spanner-activerecord/blob/b083a8750babc2653fd4dc3b2ffa7ced47bf41c6/examples/snippets/mutations/application.rb#L19).
4. Also let the user turn off the use of mutations in auto-commit mode. Dbapi only accepts SQL strings as input, and this implementation tries to 'parse' the SQL string and convert it to a mutation in a relatively simple way. If the SQL string contains special characters, comments, etc. this parsing would currently fail.